tests/MartinGeorgiev/Doctrine/DBAL/Types/BooleanArrayTest.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\MartinGeorgiev\Doctrine\DBAL\Types;
+
+use Doctrine\DBAL\Platforms\PostgreSQL94Platform;
+use MartinGeorgiev\Doctrine\DBAL\Types\BooleanArray;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+
+class BooleanArrayTest extends TestCase
+{
+    /**
+     * @var PostgreSQL94Platform
+     */
+    private $platform;
+
+    /**
+     * @var BooleanArray|MockObject
+     */
+    private $fixture;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->platform = new PostgreSQL94Platform();
```

Comment:
```suggestion
        $this->platform = $this->createMock(AbstractPlatform::class);
```

Reply:
The new `BooleanArray` type leverages `PostgreSQL94Platform::convertBooleansToDatabaseValue` and `PostgreSQL94Platform::convertFromBoolean`.

If we make this change, we have to mock these two methods increasing tests complexity and adding a third parameter to `BooleanArrayTest::validTransformations` data provider to provide the platform value.

Please, confirm that it is what you want. Thanks.

Reply:
I've implemented this in a separate commit, so you can evaluate it.

Reply:
> The new `BooleanArray` type leverages `PostgreSQL94Platform::convertBooleansToDatabaseValue` and `PostgreSQL94Platform::convertFromBoolean`.
> 
> If we make this change, we have to mock these two methods increasing tests complexity and adding a third parameter to `BooleanArrayTest::validTransformations` data provider to provide the platform value.
> 
> Please, confirm that it is what you want. Thanks.

Ah, I did not spot this earlier. Thanks for explaining your reasoning - it makes sense to me now.